c/datasetjson.c

```diff
@@ -1671,7 +1668,130 @@ void respondWithHLQNames(HttpResponse *response, MetadataQueryCache *metadataQue
   finishResponse(response);     
 #endif /* __ZOWE_OS_ZOS */
 }
+//new function
+
+static int getDSCB(char* datasetName, char* dscb, int bufferSize){
+  if (bufferSize < INDEXED_DSCB){
+    zowelog(NULL, LOG_COMP_ID_MVD_SERVER, ZOWE_LOG_WARNING, 
+            "DSCB of size %d is too small, must be at least %d", bufferSize, INDEXED_DSCB);
+    return 1;
+  }
+  Volser volser = {0};
+  DatasetName dsn = {0};
+  memset(dsn.value, ' ', DATASET_PATH_MAX);
+  
+  int lParenIndex = indexOf(datasetName, strlen(datasetName),'(',0);
+  
+  //String parsing operation, gets rid of of 3 initial unnecessary characters-> //'
+  if (lParenIndex > 0){
+    memcpy(dsn.value,datasetName+3,lParenIndex-3);
+  }
+  //String parsing operation, gets rid of of 3 initial unnecessary characters-> //'
+  else{
+    memcpy(dsn.value,datasetName+3,strlen(datasetName)-4);
+  }
+    
+  int volserSuccess = getVolserForDataset(&dsn, &volser);
+  
+  if(!volserSuccess){
+    
+    int rc = obtainDSCB1(dsn.value, sizeof(dsn.value),
+                           volser.value, sizeof(volser.value),
+                           dscb);
+
+    if (rc == 0){
+      if (DSCB_TRACE){
+        zowelog(NULL, LOG_COMP_ID_MVD_SERVER, ZOWE_LOG_WARNING, "DSCB for %.*s found\n", sizeof(dsn.value), dsn.value);
+        dumpbuffer(dscb,INDEXED_DSCB);
+      }
+    }
+    return 0;
+  }
+  else {
+    return 1;
+  }
+}
 
+void newDatasetMember(HttpResponse* response, char* datasetPath, char* memberName) {
+  char dscb[INDEXED_DSCB] = {0};
+  int bufferSize = sizeof(dscb);
+  if (getDSCB(datasetPath, dscb, bufferSize) != 0) {
+    respondWithJsonError(response, "Error decoding dataset", 400, "Bad Request");
+  }
+  else {
+    if (!isPartionedDataset(dscb)) {
+      respondWithJsonError(response, "Dataset must be PDS/E", 400, "Bad Request");
+    }
+    else {
+      char *overwriteParam = getQueryParam(response->request,"overwrite");
+      int overwrite = !strcmp(overwriteParam, "true") ? TRUE : FALSE;
+      char fullPath[DATASET_MEMBER_MAXLEN + 1]; //Max dataset path length
+      int apostrophePos = lastIndexOf(datasetPath, DATASET_PATH_MAX, '\''); //index of apostrophe
+      nullTerminate(datasetPath, apostrophePos); // remove last apostrophe by null termination
+      //concatenates dataset name with member name
+      snprintf(fullPath, DATASET_MEMBER_MAXLEN, "%s(%s)'", datasetPath, memberName);
+      FILE* datasetExists = fopen(fullPath,"r");
```

Comment:
You are really checking for the member here, right? We know the dataset exists because it has a dscb. This boolean-like name is a bit confusing considering what you are actually trying to achieve here.

I’m still concerned about the implications of using c standard functions “fopen”, “fclose”, etc. I don’t think they will compile in METAL C. Sean said it was OK to use them though.